optaweb-vehicle-routing-docs/src/main/asciidoc/development-guide.adoc

```diff
@@ -0,0 +1,253 @@
+To run backend in development mode, enter:
+
+[source,shell]
+----
+mvn spring-boot:run
+----
+
+=== Automatic restart
+
+https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#using-boot-devtools-restart[Automatic restart]
+is provided by Spring Boot DevTools and only works when the backend is running using <<run-spring-boot-maven-plugin, Spring Boot Maven Plugin>>.
+It scans files on the classpath, so you only need to recompile your changes to trigger application restart.
+No IDE configuration is needed.
+
+If your IDE has a compile-on-save feature (Eclipse, NetBeans), you just need to save the files that have changed since last compilation.
+
+IntelliJ IDEA saves changes automatically and you need to use either Recompile action, which recompiles the file in active tab, or Build Project action which recompiles all changes.
+See https://www.jetbrains.com/help/idea/compiling-applications.html[Compile and build applications with IntelliJ IDEA].
+
+=== Running backend from IntelliJ IDEA
+
+. Run `org.optaweb.vehiclerouting.OptaWebVehicleRoutingApplication` from Project window.
+This will create a Run Configuration that you will edit in the next step.
+A failure on the first run is expected due to wrong working directory.
+
+. Select menu:Run[Edit Configurations...] and then select menu:Spring Boot[OptaWebVehicleRoutingApplication].
+
+. Set *Program arguments* to `--spring.profiles.active=local` to activate the Spring profile called `local`.
+This will make the application use configuration from `application-local.properties`.
+
+. Change *Working directory* to the backend module (`optaweb-vehicle-routing-backend`).
+
+. Optionally, set *On Update action* to *Hot swap classes and update trigger file if failed*.
+This will allow you to use the Update action to quickly restart the application.
++
+See https://blog.jetbrains.com/idea/2018/04/spring-and-spring-boot-in-intellij-idea-2018-1/[Spring and Spring Boot in IntelliJ IDEA 2018.1]
+for more details.
+
+[[backend-configuration]]
+=== Configuration
+
+There are multiple ways how to set configuration properties.
+When running locally, you'll probably want to use one of these:
+
+* Set it using `application.properties` (you'll find it under `/src/main/resources/`).
+* A command line argument when running the packaged application (e.g. `java -jar optaweb-vehicle-routing-backend.jar --app.my-property=value1`).
+* An environment variable when running the application with `spring-boot:run`.
++
+For example: `app_my_property=value1 ./mvnw spring-boot:run` if the property name is `app.my-property` (this requires
+https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-features-external-config-relaxed-binding[relaxed binding]
+which only works if the property is defined using `@ConfigurationProperties`).
+
+[NOTE]
+It's not possible to set properties simply via `-D` when running the application using Spring Boot Maven plugin (`./mvnw spring-boot:run -Dmy-property`).
```

Comment:
It is not possible to set properties by specifying -D when...using the Spring Boot....